### DISCUSSION TITLE
Run Task Freezes for some Variable References

### DISCUSSION THREAD

**Discussion by shengchao-lin** [microsoft/vscode-cpptools#10686] [Q&A]
When variable reference is used that relates to the current file, e.g., `${file}`, `${fileBasename}`, etc, the run task freezes. However, a simple echo does not. See the example below. Task "My C++ build" hangs but "echo" does not.

When freezing, the terminal tab shows:
<img width="228" alt="Screenshot 2023-03-15 at 10 37 47 PM" src="https://user-images.githubusercontent.com/43042885/225495941-e41fbaed-d21b-47b8-9883-661cab8446e0.png">

If task "echo" is ran first, launched the terminal tab, and ran successfully, then task "My C++ build" can ran successfully too.

Additionally, when the `"${workspaceFolder}/${fileBasename}"` is replaced with something like `"${workspaceFolder}/main.cpp"`, the task "My C++ build" runs successfully.

What could be the issue here? 
```
"tasks": [
{
	"type": "cppbuild",
	"label": "build",
	"command": "g++",
	"args": [
		"-fdiagnostics-color=always",
		"-g",
		"${workspaceFolder}/${fileBasename}",
		"-o",
		"${workspaceFolder}/output",
		"-std=c++20"
	],
	"detail": "compiler: /usr/bin/g++",
	"group": {
		"kind": "build",
		"isDefault": true
	}
},
{
	"type": "shell",
	"label": "echo",
	"command": "echo ${workspaceFolder}/${fileBasename}",
}
]

```

Platform: MacOS Ventura 13.2.1
VSCODE: 
  Version: 1.76.2
  Commit: ee2b180d582a7f601fa6ecfdad8d9fd269ab1884
  Date: 2023-03-14T17:54:09.061Z
  Electron: 19.1.11
  Chromium: 102.0.5005.196
  Node.js: 16.14.2
  V8: 10.2.154.26-electron.0
  OS: Darwin x64 22.3.0
  Sandboxed: No

Extension used: C/C++

**Comment by sean-mcmanus** [comment]
What version of the C/C++ extension?

**Reply by shengchao-lin** [reply]
Oops, forgot. It's v1.14.4.
